tests/page-objects/lending-bar-auto-renew.ts

```diff
@@ -0,0 +1,307 @@
+import { type Page, Locator, expect } from '@playwright/test';
+
+export class LendingBarAutoRenew {
+  readonly page: Page;
+
+  readonly iaBookActions: Locator;
+  readonly demoControls: Locator;
+
+  readonly loanSecondsFor5Min: Number;
+  readonly loanSecondsFor10Min: Number;
+  readonly loanSecondsFor30Min: Number;
+  readonly loanSecondsFor60Min: Number;
+
+  private timeInBrowser: Date;
+
+  public constructor(page: Page) {
+    this.page = page;
+
+    this.iaBookActions = this.page.locator('ia-book-actions');
+    this.demoControls = this.page.locator('.initial');
+
+    this.loanSecondsFor5Min = 300; // seconds
+    this.loanSecondsFor10Min = 600; // seconds 
+    this.loanSecondsFor30Min = 1800; // seconds
+    this.loanSecondsFor60Min = 3600; // seconds
+
+    this.timeInBrowser = new Date('2024-07-08T11:00:00'); // time
+  }
+
+  async clickOnBrowsedButton() {
+    const browsedCheckbox = this.demoControls.getByText('user_has_browsed', { exact: true });
+    await browsedCheckbox.click();
+  }
+
+  async gotoPage(uri: string) {
+    await this.page.goto(uri, { waitUntil: 'networkidle' });
+
+    this.timeInBrowser = await this.page.evaluate(() => {
+      // return new Date().toISOString();
+      return new Date('2024-07-08T11:00:00');
+    });
+    console.log('timeInBrowser:', this.timeInBrowser);
+
+    await this.page.clock.setSystemTime(this.timeInBrowser.toString());
+    await this.page.waitForTimeout(1000);
+  }
+
+  async getTimerCountdownSeconds() {
+    const timerElement = this.page.locator('timer-countdown .second');
+    if (timerElement) return Number(await timerElement.textContent());
+    return false;
+  }
+
+  async getIncrementedTime(Minutes: number) {
+    const timeObj = new Date(this.timeInBrowser.toString());
+    const incresedTime = new Date(timeObj.getTime() + Minutes * 60000);
+    console.log('originalTime: ', timeObj)
+    console.log('incresedTime: ', incresedTime);
+    return incresedTime;
+  }
+
+  async warningModelVisible() {
+    await expect(await this.page.locator('.backdrop')).toBeVisible();
+    await expect(await this.page.getByText('Are you still reading?')).toBeVisible();
+  }
+
+
+  async warningModelHidden() {
+    await expect(await this.page.locator('.backdrop')).toBeHidden();
+    await expect(await this.page.getByText('Are you still reading?')).toBeHidden();
+  }
+
+  // auto renew feature testing starts from here....
+  /**
+   * 5 minutes loan tests, the timer details are as follow:-
+   * - loanTotalTime: 300,
+   * - loanRenewAtLast: 240,
+   * - pageChangedInLast: 60
+   */
+  async autoRenew_userJustBrowsedFor5Min() {
+    await this.clickOnBrowsedButton();
+
+    // install clock timer is -- 300000ms / 300s / 5min
+    await this.page.clock.install({ time: await this.getIncrementedTime(5) });
+
+    await this.page.clock.runFor('01:20'); // (MM:SS)
+    await this.page.waitForTimeout(1000);
```

Comment:
QUESTION: Again, why these constant timeouts? If they are necessary "padding", I wonder if it would be helpful to DRY these up and use a helper function like `this.page.waitForClockRunStart()` or something similar, to explain the need for the delay. (Then, if you ever need to change the delay value, it'll be in just one place instead of everywhere.)